client/internal/engine.go

```diff
@@ -134,6 +179,29 @@ func (e *Engine) Start() error {
 		return err
 	}
 
+	muxConnPort := 55050
+	if e.config.UDPMuxPort != nil {
```

Comment:
if UDPMuxPort defaults to 0, we can simply accept its value and we will have either a random port from 0 or wherever we set as UDPMuxPort's value.